doc/Overview/Mvux/Advanced/Commands.md

```diff
@@ -0,0 +1,191 @@
+
+    ```csharp
+    public void DoWork();
+    public ValueTask DoWork();
+    public ValueTask DoWork(CancellationToken ct);    
+    ```
+
+    A `DoWork` command will be generated in the Bindable Proxy Model. When that command is executed via a button-click etc., the method in the Model will be called:
+
+    ```xml
+    <Button Command="{Binding DoWork}" />
+    ```
+
+    As you may have noticed, the method can be asynchronous and take a `CancellationToken`, but these are not mandatory and commands will be generated for synchronous methods as well.
+
+    ### Adding command parameters
+
+    An additional parameter can be added to the method, which is then assigned with the value of the `CommandParameter` received from the View. For instance, when using a Button and clicking it, the method will be called with the [`Button.CommandParameter`](https://learn.microsoft.com/en-us/windows/windows-app-sdk/api/winrt/microsoft.ui.xaml.controls.primitives.buttonbase.commandparameter) value, given that the type of the value matches the method parameter type. Otherwise the command's `CanExecute` will be false thereby disabling the button:
+
+    ```csharp
+    public void DoWork(int param)
+    {
+        ...
+    }
+    ```
+
+    ```xml
+    <Button Command="{Binding DoWork}" CommandParameter="{Binding MyIntegerValue}" />
+    ```    
+
+    Accordingly, if the `CommandParameter` type does not match the parameter type of the method, the button will remain disabled:
+
+    ```xml
+    <Button Command="{Binding DoWork}" CommandParameter="A string, not an integer" />
+    ```
+
+    Result:
+
+    ![A disabled refresh button](../Assets/DisabledButton.jpg)
+        
+    ### Command generation rules:
+    
+     - Can be either synchronous or asynchronous (i.e. `public void`, `public async ValueTask`, or `public async Task`)
+     - Any parameter that has a type and name matching a Feed or a State in this Model will be evaluated when the Command is invoked and its current latest value will be passed in as an argument.  
+        You'll find this feature to be very powerful in invoking commands by combining data from various Feeds in addition to a command parameter received from the View.
+
+        For example:
+
+        ```csharp
+        public IFeed<int> CounterValue => ...
+
+        public void ResetCounter(int counterValue)
+        {
+            ...
+        }
+        ```
+
+        When the command is executed, because the parameter name `counterValue` matches a feed name in the Model, this parameter will be materialized with the actual value from the Feed when this method is called on command execution.  
+        This behavior can be controlled and configured using the [`FeedParameter`](#feedparameter-attribute) and [`ImplicitFeedCommandParameter`](#implicitfeedcommandparameter-attribute) attributes.
+
+     - Can have one `CancellationToken` as its last parameter, but it's not mandatory.
```

Comment:
What can cause the command to be cancelled?

Reply:
I've spoken to @dr1rrb about this.
Also have a look at [this](https://github.com/unoplatform/uno.extensions/discussions/1363).

Reply:
Currently the main reason is the model being disposed (which will dispose the command). And anyway you know my opinion about this: we should never start an asynchronous operation without a `CancellationToken` :) (Even if the operation ignores it at some point to ensure coherence of the state)

Reply:
The question is if there's a more integrated way to cancel commands.

Reply:
I was wondering whether there's any circumstances where the command is cancelled - @dr1rrb has answered this as it gets cancelled when the model is disposed. 
@dr1rrb does the command also get cancelled if the user clicks a button repeatedly? Perhaps something to consider in future but I'm not sure if/how this would be configurable

Reply:
@nickrandolph when the user clicks the button the command goes into executable mode which is meanwhile unexecutable - it appears disabled.

Reply:
This would require a two-state button where it has a configuarable cancellation text and acts as a cancellation button when its command is executing.